Ubuntu server doc V1 (#9)
EpiMac/docs: docs/X86_Silicon/index.mdx

### Télécharger l'ISO d'un OS

Pour cette documentation nous utiliserons l'image ISO d'un Ubuntu Server avec une architecture x86_64 similaire au architecture utilisée sur les machines de l'EPITA.

[https://ubuntu.com/download/server](https://ubuntu.com/download/server)

*Prenez une version LTS pour vous assurez d'avoir une version stable.*

### Créer une machine virtuelle émulée

Sur UTM, cliquez sur le bouton `Create a new Virtual Machine`. Ensuite `Emulate`, puis `Other`.

Vous pouvez ensuite Ajouter (en cliquant sur `Browse`) votre chemin vers l'image ISO que vous venez de télécharger.

Cliquez sur `Continue`.

### Configurer la machine virtuelle

- Vérifiez que vous avez bien sélectionné `x86_64` dans la liste déroulante `Architecture`.

- Laissez `System` sur le choix par défaut.

- Vous pouvez ensuite choisir la quantité de RAM que vous souhaitez allouer à votre machine virtuelle. 6144 MB et 6 coeurs est recommandée.

<img src="/img/doc_X86/doc_X86_Settings.png" alt="Menu" width="400" style={{borderRadius: 5}} />

- Vous pourrez ensuite spécifier la taille allouée pour votre machine virtuelle. Séléctionnez 64GB.

- Vous pouvez ignorer la partie `Shared Directory`.

> - Sur le sommaire, vous pouvez nommez la VM comme vous le souhaitez.

Reviewer: ```suggestion
- Sur le sommaire, vous pouvez nommer la VM comme vous le souhaitez.
```

### Installation de la machine virtuelle

> - Lancer votre machine virtuelle.

Reviewer: ```suggestion
- Lancez votre machine virtuelle.
```

*Vous devriez voir une fenêtre s'ouvrir.*
:::caution
Utilisez la touche TAB, Espace et Enter pour naviguer dans le menu.

> :::

Reviewer: Mets un saut de ligne avant et après la ligne 66, parfois Docusaurus parse mal les admonitions

- Prenez le premier choix de la liste affichée.

*Votre VM devrait s'installer avec des **`[ OK ]`** sur la gauche*

*Vous devriez ensuite voir une fenêtre avec differentes langues*

- Choississez une langue.

> - Ensuite, si une mise à jour est disponible vous auriez le choix de l'installer.

Reviewer: ```suggestion
- Ensuite, si une mise à jour est disponible vous pouvez l'installer si vous le souhaitez
```

> - Ensuite, sélectionnez le bon clavier.

Reviewer: ```suggestion
- Ensuite, sélectionnez le bon layout pour votre clavier.
```

- Ensuite, vous aurez la possiblité de selectionner le type d'installation. Choississez la premier option `Ubuntu Server`.

- Ensuite, vous serez sur l'interface Réseau. Votre réseau devrait être automatiquement selectionné. Faites suivant.

> 

Reviewer: Ca fait beaucoup de `Ensuite`, peut être varier ^^

---
sidebar_position: 2
title: x86_64 on Apple Silicon
tags:
  - VM
  - MacOS
  - GDB
---

> #### How do I get a WSL on Mac Silicon?

Reviewer: Windows Subsystem for Linux ?

This documentation aims to teach you how to develop on x86_64 architecture with your Silicon Mac.

> 

Reviewer: Add a paragraph to explain the purpose of the tutorial and why it could be interesting (its applications for example)

## Prerequisites

> - A Silicon Mac,

Reviewer: ```suggestion
- A Silicon Mac (M1/M2/M3)
```

- A minimum of 64 GB free disk space.

## Download UTM

UTM lets you virtualize and emulate any architecture on a Silicon Mac.

Download the latest UTM version:

> [Cliquez ici](https://github.com/utmapp/UTM/releases/latest/download/UTM.dmg)

Reviewer: ```suggestion
[Download here](https://github.com/utmapp/UTM/releases/latest)
```
Ne pas mettre un téléchargement automatique mais laisser les gens télécharger par eux même.

Note that there is a paid version of UTM on the App Store, identical to the one above except for automatic application updates.

> ## Download OS ISO

Reviewer: ```suggestion
## Download a linux distribution for our VM
```

> For this documentation, we'll be using the ISO image of an Ubuntu Server with an x86_64 architecture similar to the one used on EPITA computers.
> 
> [Link to ISO image](https://ubuntu.com/download/server)
> 

Reviewer: ```suggestion
For this tutorial, we'll be using an Ubuntu Server image, on x86_64 architecture which is similar to the one used on EPITA computers. You can find the ISO image [here](https://ubuntu.com/download/server).

:::note

You can choose the distribution you want, as long as you take a x86_64 architecture

:::
```

> *Take an LTS version to ensure you have a stable version.*

Reviewer: ```suggestion
```

## Creating an emulated virtual machine

On UTM, click on the `Create a new Virtual Machine` button. Then `Emulate`, then `Other`.

You can then add (by clicking on `Browse`) your path to the ISO image you've just downloaded.

Click on `Continue`.

## Setting up the virtual machine

- Check that you have selected `x86_64` in the `Architecture` drop-down list.

- Leave `System` as the default.

- You can then choose the amount of RAM you wish to allocate to your virtual machine. 6144 MB and 6 cores is recommended.

<img src="/img/doc_X86/doc_X86_Settings.png" alt="Menu" width="400" style={{borderRadius: 5, boxShadow: '0 0 10px rgba(0,0,0,0.5)', display:"block", marginLeft: "auto", marginRight: "auto", marginTop: 30, marginBottom: 30}} />

- You can then specify the size allocated for your virtual machine. Select 64GB.

- You can ignore the Shared Directory section.

- On the summary, you can name the VM as you wish.

## Virtual machine installation

- Launch your virtual machine.

### Basic VM configuration

*You should see a window open.*

> :::caution Caution

Reviewer: ```suggestion
:::warning
```

Je vais passer la doc à docusaurus v3, `caution` passe en deprecated

- You're now on the Network interface. Your network should be automatically selected. You can proceed.

- Select next on the `Proxy` tab.

*You are now on `Ubuntu Archive Mirror`. Wait for it to detect the archives*.

- If you get no error, do next.

### Advanced VM configuration

- Then you can choose the VM partitions. Do `Use entire disk`. You should see `local disk 64GB`.

*You can encrypt your disk if you wish.*

- Do next and `Continue` on the `Storage Configuration` tab.

- You can now add your VM information.

![Menu](/img/doc_X86/doc_X86_Menu.png)

- Click on next for the `Ubuntu Pro` tab.

*You're now on the `SSH Setup` tab*.

- Press space to check `SSH server` installation.

*You are now on the `Featured Server Snaps`* tab.

- Click next.

*You should see a window with `Installing the system`.*

:::info
The installation process is a bit slow...
:::

- You can then click on `Reboot Now`.

- When you see the error message `Unable to unmount`, stop your VM. Then go to the UTM interface and click on CD/DVD. Then click on `Clear`.

- You can restart your VM and connect to your new VM.

![VM preview](/img/doc_X86/doc_X86_VM.png)

> ## Bonus: Develop on x86_64 your projects with VS Code.

Reviewer: ```suggestion
## Connect VS Code to your VM
```

> - Add an SSH key to your VM. (You can use the CRI tutorial to add an SSH key to your VM).

Reviewer: ```suggestion
- Add an SSH key to your VM.
```
1) C'est la Forge
2) On a aussi des étudiants non épitéens

Je pense qu'il vaut mieux éviter de mettre cette info

- Stop your VM.

- On UTM, click on `Edit` then `Network`.

- Change the `Network Mode` field to `Emulated VLAN`.

- You'll see `Port Forward` in the navigation bar. Click on it.

- Enter the following information:

![Port Forward](/img/doc_X86/doc_X86_Clear.png)

- Install the `Remote Explorer` extension on VS Code: [Click here](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-explorer)

- You'll see `Remote Explorer` in the navigation bar. Click on it.

- Add a new SSH connection with field `ssh <username>@127.0.0.1 -p 2200`. Replace `<username>` with the `server's name` field entered in the menu.

- You'll need to enter your password to connect to your VM.

- You now have access to your VM's terminal from VS Code and can compile your projects with an x86_64 architecture!

> :::tip Tip

Reviewer: ```suggestion
:::tip
```

> You can now use gdb, and all the tools that are not usually available on a Silicon computer.

Reviewer: Ce n'est pas en théorie un défaut vu que Apple utilise un autre outil similaire (lldb)
